src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVUtils.java

```diff
@@ -19,6 +27,81 @@
 
     public static final String GATKSV_CONTIG_ALIGNMENTS_READ_GROUP_ID = "GATKSVContigAlignments";
 
+    public static String getSampleId(final SAMFileHeader header) {
+        final List<SAMReadGroupRecord> readGroups = header.getReadGroups();
+        final Set<String> sampleSet = readGroups.stream().map(SAMReadGroupRecord::getSample).collect(Collectors.toSet());
+
+        Utils.validate(sampleSet.size() == 1,
+                "Read groups must contain reads from one and only one sample, " +
+                        "but we are finding the following ones in the given header: \t" + sampleSet.toString());
+
+        final String sample = sampleSet.iterator().next();
+        return sample;
+    }
+
+    /**
+     * Given {@code sortOrder}, provide appropriate comparator.
+     * Currently only support coordinate or query-name order,
+     * and throws UserException if other values are specified.
+     */
+    public static SAMRecordComparator getSamRecordComparator(final SAMFileHeader.SortOrder sortOrder) {
+        final SAMRecordComparator samRecordComparator;
+        switch (sortOrder) {
+            case coordinate:
+                samRecordComparator = new SAMRecordCoordinateComparator();
+                break;
+            case queryname:
+                samRecordComparator = new SAMRecordQueryNameComparator();
+                break;
+            default:
+                throw new UserException("Unsupported assembly alignment sort order specified");
+        }
+        return samRecordComparator;
+    }
+
+    /**
+     * todo: this should be fixed in a new major version of htsjdk
+     * this exist because for whatever reason,
+     * VC.getAttributeAsStringList() sometimes returns a giant single string, while using
+     * VC.getAttributeAsString() gives back an array.....
+     */
+    public static List<String> getAttributeAsStringList(final VariantContext vc, final String attributeKey) {
+        return getAttributeAsStringStream(vc, attributeKey)
+                .collect(Collectors.toList());
+    }
+
+    public static Stream<String> getAttributeAsStringStream(final VariantContext vc, final String attributeKey) {
+        if ( ! vc.hasAttribute(attributeKey) )
+            return Stream.empty();
+        return vc.getAttributeAsStringList(attributeKey, "").stream()
+                .flatMap(s -> {
+                    if ( s.contains(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR) ) {
+                        return Arrays.stream( s.split(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR) );
+                    } else {
+                        return Stream.of(s);
+                    }
+                });
+    }
+
+    public static SimpleInterval makeOneBpInterval(final String chr, final int pos) {
+        return new SimpleInterval(chr, pos, pos);
```

Comment:
Maybe worth adding comment noting that SimpleIntervals are closed instead of semi-closed. I personally am occasionally confused by this difference.

Reply:
I think that's standard: `SimpleInterval`'s, inherited from `Locatable`'s from htsjdk, are 1-based, fully closed intervals. I learned this by doing some work on htsjdk.

Reply:
Added more doc for this method.